plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2347,6 +2347,9 @@ private HiveColumnHandle createProjectedColumnHandle(HiveColumnHandle column, Li
         HivePartitioningHandle leftHandle = (HivePartitioningHandle) left;
         HivePartitioningHandle rightHandle = (HivePartitioningHandle) right;
 
+        if (leftHandle.isUsePartitionedBucketing() != rightHandle.isUsePartitionedBucketing()) {
+            return Optional.empty();
```

Comment:
It looks like it fixes a bug. A separate PR for release notes' sake?

Reply:
No need for separate release note as the PR introducing this feature will be part of next release. It would be good to land this PR though.

I though it's a bug, but `arguments` in `io.trino.sql.planner.Partitioning#isCompatibleWith` don't match (bucketed vs bucketed and partitioned columns). However, explicit check is better